includes/data-stores/class-wc-product-data-store-cpt.php

```diff
@@ -680,7 +680,7 @@ protected function update_version_and_type( &$product ) {
 	 */
 	protected function clear_caches( &$product ) {
 		wc_delete_product_transients( $product->get_id() );
-		wp_cache_delete( 'product-' . $product->get_id(), 'products' );
+		wp_cache_delete( 'object-' . $product->get_id(), 'products' );
```

Comment:
The intention is that object- prefix is used on the key, but the group is what gets cleared.